src/python/tests/test_configuration_manager.py

```diff
@@ -0,0 +1,653 @@
+                                  category_value={
+                                      'info': {
+                                          'description': 'INTEGER type with default 1',
+                                          'type': 'INTEGER',
+                                          'default': '1'}})
+        assert ('ValueError: Invalid entry_val for entry_name "type" for item_name info. valid: ' +
+                "['boolean', 'integer', 'string', 'IPv4', " +
+                "'IPv6', 'X509 certificate', 'password', 'JSON']") in str(error_exec)
+
+    async def test_create_category_invalid_entry_value_for_type(self):
+        """
+        Test case where value is set to the actual "value" rather than the string of the value
+        :Assert:
+            Assert TypeError when type is set to bool rather than 'boolean'
+        """
+        with pytest.raises(TypeError) as error_exec:
+            await create_category(category_name='boolean', category_description='boolean type',
+                                  category_value={'info': {
+                                      'description': 'boolean type with default False',
+                                      'type': bool,
+                                      'default': 'False'
+                                  }})
+        assert ("TypeError: entry_val must be a string for item_name " +
+                "info and entry_name type") in str(error_exec)
+
+    async def test_create_category_invalid_entry_value_for_default(self):
+        """
+        Test case where value is set to the actual "value" rather than the string of the value
+        :Assert:
+            Assert TypeError when default is set to False rather than 'False'
+        """
+        with pytest.raises(TypeError) as error_exec:
+            await create_category(category_name='boolean',
+                                  category_description='boolean type',
+                                  category_value={'info': {
+                                      'description': 'boolean type with default False',
+                                      'type': 'boolean',
+                                      'default': False
+                                  }})
+        assert ("TypeError: entry_val must be a string for item_name " +
+                "info and entry_name default") in str(error_exec)
+
+    async def test_create_category_invalid_entry_value_for_description(self):
+        """
+        Test case where value is set to the actual "value" rather than the string of the value
+        :Assert:
+            Assert TypeError when description is set to None rather than ''
+        """
+        with pytest.raises(TypeError) as error_exec:
+            await create_category(category_name='boolean',
+                                  category_description='boolean type',
+                                  category_value={'info': {
+                                      'description': None,
```

Comment:
is empty string "" allowed?

Reply:
As long as the information for `category_value` is quoted (""), the code will accept it. As such, I didn't see a reason to add such a test; though it's not a problem to add it.